assembl/views/discussion/views.py

```diff
@@ -51,7 +51,9 @@ def get_default_context(request):
     base = base_default_context(request)
     slug = request.matchdict['discussion_slug']
     try:
-        discussion = Discussion.default_db.query(Discussion).filter(Discussion.slug==slug).one()
+        discussion = Discussion.default_db.query(Discussion).filter(Discussion.slug==slug).first()
+        if not discussion:
+            discussion = find_discussion_from_slug(request, slug)
```

Comment:
Instead of those 3 lines, it should really just be:
`discussion = find_discussion_from_slug(request, slug)`
and put this logic in `find_discussion_from_slug` if necessary.